pkg/common/nats/conn.go

```diff
@@ -74,6 +79,34 @@ func NewNatsConn(ctx context.Context, config commonconfig.EventingNatsConfig) (*
 		opts = append(opts, o)
 	}
 
+	// reconnection options
+	if config.ConnOpts.RetryOnFailedConnect {
+		reconnectWait := config.ConnOpts.ReconnectWait * time.Millisecond
+		logger.Infof("Configuring retries: %#v", config.ConnOpts)
+		opts = append(opts, nats.RetryOnFailedConnect(config.ConnOpts.RetryOnFailedConnect))
+		opts = append(opts, nats.ReconnectWait(reconnectWait))
+		opts = append(opts, nats.MaxReconnects(config.ConnOpts.MaxReconnects))
+		opts = append(opts, nats.CustomReconnectDelay(func(attempts int) time.Duration {
+			if (config.ConnOpts.MaxReconnects - attempts) < 0 {
+				logger.Fatalf("Failed to recconect to Nats, not attempts left")
+			}
+			if attempts < 5 {
+				logger.Infof("Attempts left: %d", config.ConnOpts.MaxReconnects-attempts)
+			}
```

Comment:
Why is 5 hardcoded here?

Reply:
This is just to show 5 last messages or recconnecting attepmts. Should I give it a name, make it debug and log everything or just remove it?

Reply:
fixed